torch/linalg/__init__.py

```diff
@@ -248,42 +268,60 @@
 eig = _add_docstr(_linalg.linalg_eig, r"""
 linalg.eig(input, *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalues and eigenvectors of a square
-matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+Computes the eigenvalue decomposition of a square matrix or batch of square matrices.
+
+For a square matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) V⁻¹`, where `V⁻¹` is the inverse of `V`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{-1}
+
+This decomposition exists if :math:`A` is diagonalizable (for example, when all its eigenvalues are different).
```

Comment:
what does "diagonalizable" mean?

Reply:
what happens if the decomposition doesn't exist?

Reply:
That's a tricky one, as it is defined as "matrices for which this function would succeed" :D That's why I gave there a simple condition that works in almost any practical situation: A matrix with different eigenvalues is diagonalisable.

Would it be fine to link to the following wikipedia article?
https://en.wikipedia.org/wiki/Diagonalizable_matrix#Definition

If the decomposition does not exist, then the eigenvalues returned by this function are incorrect. The point to be made here is that this is not a problem of the algorithm, but a problem intrinsical to the matrix.

Reply:
I have updated this point and the warning below. How does it read now?